File: base/src/main/java/de/whitefrog/frogr/exception/DuplicateEntryException.java
package de.whitefrog.frogr.exception;

import de.whitefrog.frogr.model.Base;

import java.lang.reflect.Field;

/**
 * Thrown when a duplicate entry would be generated during persistence.
 */
public class DuplicateEntryException extends PersistException {
  private Base bean;
  private Field field;
  
  public DuplicateEntryException(String message, Base bean) {
    super(message);
    this.bean = bean;
  }

  public DuplicateEntryException(String message, Base bean, Field field) {
    super(message);
    this.bean = bean;
    this.field = field;
  }

  public Base getBean() {
    return bean;
  }
  
  public Field getField() {
    return field;
  }
}

File: base/src/main/java/de/whitefrog/frogr/exception/DuplicateEntryException.kt
package de.whitefrog.frogr.exception

import de.whitefrog.frogr.model.Base

import java.lang.reflect.Field

/**
 * Thrown when a duplicate entry would be generated during persistence.
 */
class DuplicateEntryException : PersistException {
  var bean: Base? = null
  var field: Field? = null

  constructor(message: String?, bean: Base) : super(message) {
    this.bean = bean
  }

  constructor(message: String?, bean: Base, field: Field) : super(message) {
    this.bean = bean
    this.field = field
  }
}
